### PR TITLE
Reformat jit source base using clang-format and clang-tidy

### PR DESCRIPTION
**Pull request by adiaaida**
This change is the result of running clang-tidy and clang-format on the jit source base. More details on proposed changes can be found in #6034.

### CODE REVIEW

File: src/jit/alloc.cpp [hunk 1]
```diff
@@ -81,11 +83,8 @@ size_t ArenaAllocator::getDefaultPageSize()
 // ArenaAllocator::ArenaAllocator:
 //    Default-constructs an arena allocator.
 ArenaAllocator::ArenaAllocator()
-    : m_memoryManager(nullptr)
-    , m_firstPage(nullptr)
-    , m_lastPage(nullptr)
-    , m_nextFreeByte(nullptr)
-    , m_lastFreeByte(nullptr)
+    : m_memoryManager(nullptr), m_firstPage(nullptr), m_lastPage(nullptr), m_nextFreeByte(nullptr),
```
Comment (mikedn) [inline src/jit/alloc.cpp]: This looks like a step backwards to me, it's easier to compare the initialization list with data members definitions to see if anything is missing if there's one initialization per line.

File: src/jit/alloc.cpp [hunk 2]
```diff
@@ -115,16 +111,16 @@ ArenaAllocator& ArenaAllocator::operator=(ArenaAllocator&& other)
     assert(!isInitialized());
 
     m_memoryManager = other.m_memoryManager;
-    m_firstPage = other.m_firstPage;
-    m_lastPage = other.m_lastPage;
-    m_nextFreeByte = other.m_nextFreeByte;
-    m_lastFreeByte = other.m_lastFreeByte;
+    m_firstPage     = other.m_firstPage;
```
Comment (mikedn) [inline src/jit/alloc.cpp]: Hrm, I take it that we'll keep the crazy vertical alignment that's already everywhere and make it even worse after formatting...

File: src/jit/alloc.cpp [hunk 3]
```diff
@@ -230,18 +226,18 @@ void ArenaAllocator::destroy()
     assert(isInitialized());
 
     // Free all of the allocated pages
-    for (PageDescriptor* page = m_firstPage, *next; page != nullptr; page = next)
+    for (PageDescriptor *page = m_firstPage, *next; page != nullptr; page = next)
```
Comment (mikedn) [inline src/jit/alloc.cpp]: Current CC say that the * is next to the type, not next to the variable.

File: src/jit/alloc.cpp [hunk 4]
```diff
@@ -433,13 +427,12 @@ void ArenaAllocator::shutdown()
 }
 
 PooledAllocator PooledAllocator::s_pooledAllocator;
-LONG PooledAllocator::s_pooledAllocatorState = POOLED_ALLOCATOR_NOTINITIALIZED;
+LONG            PooledAllocator::s_pooledAllocatorState = POOLED_ALLOCATOR_NOTINITIALIZED;
```
Comment (mikedn) [inline src/jit/alloc.cpp]: I have no idea why this whitespace got inserted here...

File: src/jit/bitsetasuint64.h [hunk 5]
```diff
@@ -166,23 +165,25 @@ class BitSetOps</*BitSetType*/UINT64,
...
-        const int CharsForUINT64 = sizeof(UINT64)*2;
-        char * res = NULL;
-        const int AllocSize = CharsForUINT64 + 4;
-        res = (char*)alloc->Alloc(AllocSize);
-        UINT64 bits = bs;
-        unsigned remaining = AllocSize;
-        char* ptr = res;
+        IAllocator* alloc          = BitSetTraits::GetDebugOnlyAllocator(env);
+        const int   CharsForUINT64 = sizeof(UINT64) * 2;
+        char*       res            = NULL;
```
Comment (mikedn) [inline src/jit/bitsetasuint64.h]: It's weird how 0 and NULL got replaced only in some places with nullptr.

File: src/jit/emitfmtsarm64.h [hunk 6]
```diff
@@ -3,84 +3,84 @@
...
-// enum insFormat   instruction            enum ID_OPS 
-//                  scheduling 
-//                  (unused)                       
+// enum insFormat   instruction            enum ID_OPS
+//                  scheduling
+//                  (unused)
 //////////////////////////////////////////////////////////////////////////////
 
-IF_DEF(NONE,        IS_NONE,               NONE)     //
+IF_DEF(NONE, IS_NONE, NONE) //
```
Comment (mikedn) [inline src/jit/emitfmtsarm64.h]: This looks like something that shouldn't have been reformatted

File: src/jit/bitsetasuint64.h (hunk 5 above)
Comment (adiaaida) [inline src/jit/bitsetasuint64.h]: That is a weird spot to not have it changed. I'll look into it.

File: src/jit/alloc.cpp (hunk 4 above)
Comment (adiaaida) [inline src/jit/alloc.cpp]: It's aligning it with the line above because we are using AlignConsecutiveDeclarations. It's weird, but it's what clang-format thinks it correct.

File: src/jit/alloc.cpp (hunk 3 above)
Comment (adiaaida) [inline src/jit/alloc.cpp]: clang-format has some irregularities with multiple pointers in for loops. We haven't been able to figure out a way to handle this, so we will have to live with it until it gets fixed.

File: src/jit/alloc.cpp (hunk 2 above)
Comment (adiaaida) [inline src/jit/alloc.cpp]: We are trying to align declarations.

File: src/jit/alloc.cpp (hunk 1 above)
Comment (adiaaida) [inline src/jit/alloc.cpp]: We don't have many options with clang-format. I haven't been able to find a way to force it to put them all on one line like we would like it, unfortunately.

File: src/inc/palclr.h [hunk 7]
```diff
@@ -484,7 +484,14 @@
 //  and it provides __declspec(selectany) to instruct the linker to merge
 //  duplicate external const static data copies into one.
 //  
+#if defined(__clang__)
+#define SELECTANY extern
```
Comment (janvorli) [inline src/inc/palclr.h]: What is the reason for this change? I believe Clang supports the __declspec(selectany).

**Comment by CarolEidt** [comment]
>              (curAssertion->op2.kind == O2K_CONST_DOUBLE))

I think it is better to either have all the components of a (sub)expression on the same line, OR have them each on their own line, instead of this kind of split. What is the rule that is being followed here?

---
Refers to: src/jit/assertionprop.cpp:593 in 3ce67cc. [](commit_id = 3ce67cc1dde5ba5e3879aa8d747c3b7c123dfe98, deletion_comment = False)

**Comment by CarolEidt** [comment]
>                 }

So, is it no longer allowed (as is currently the case in the coding conventions) for an if with a single then statement and no else clause, to be un-bracketed?

---
Refers to: src/jit/assertionprop.cpp:1357 in 3ce67cc. [](commit_id = 3ce67cc1dde5ba5e3879aa8d747c3b7c123dfe98, deletion_comment = False)

**Comment by CarolEidt** [comment]
>          ++ki)

Wow, this is really ugly. Can't we split these to 3 separate lines in this case?

---
Refers to: src/jit/assertionprop.cpp:1478 in 3ce67cc. [](commit_id = 3ce67cc1dde5ba5e3879aa8d747c3b7c123dfe98, deletion_comment = False)

File: src/jit/block.cpp [hunk 8]
```diff
@@ -232,38 +233,126 @@ void BasicBlock::dspBlockILRange()
...
+        printf("new[] ");
+    }
+    if (bbFlags & BBF_HAS_NEWOBJ)
+    {
+        printf("newobj ");
+    }
 #if FEATURE_EH_FUNCLETS && defined(_TARGET_ARM_)
-    if (bbFlags & BBF_FINALLY_TARGET)       printf("ftarget ");
+    if (bbFlags & BBF_FINALLY_TARGET)
+        printf("ftarget ");
```
Comment (CarolEidt) [inline src/jit/block.cpp]: So here's a weird one - all the other `if`s get braces, but not this one?

**Comment by CarolEidt** [comment]
>            oper == GT_ADD || oper == GT_SUB);

This makes it much more difficult to see the list of operators. If they have to be combined, it would be better to at least have them aligned. There's a lot of visual noise here.

---
Refers to: src/jit/codegenxarch.cpp:1435 in 3ce67cc. [](commit_id = 3ce67cc1dde5ba5e3879aa8d747c3b7c123dfe98, deletion_comment = False)

**Comment by CarolEidt** [comment]
>             CORINFO_HELP_ASSIGN_REF_EBP, CORINFO_HELP_ASSIGN_REF_ESI, CORINFO_HELP_ASSIGN_REF_EDI,

Again, IMO, this makes it more difficult to figure out how this is being initialized - especially since the two rows are broken down differently.

---
Refers to: src/jit/codegenxarch.cpp:5769 in 3ce67cc. [](commit_id = 3ce67cc1dde5ba5e3879aa8d747c3b7c123dfe98, deletion_comment = False)

File: src/jit/block.cpp (hunk 8 above)
Comment (adiaaida) [inline src/jit/block.cpp]: We have only run x64. This if is under a TARGET_ARM ifdef, so it didn't get modified due to clang-tidy running on preprocessed code.

**Comment by adiaaida** [comment]
> So, is it no longer allowed (as is currently the case in the coding conventions) for an if with a single then statement and no else clause, to be un-bracketed?

Correct. Clang-format does not have an option for not requiring braces for single line ifs with no else clause. It does have an option for allowing short if statements to be on a single line, but I don't think that is technically ok for us now, so I didn't go that route.

**Comment by adiaaida** [comment]
@CarolEidt I agree on some of the formatting changes being hard to read, but there's no support in clang-format for making some of those things more readable. I think in some of those cases, it might honestly be better for us to have the comparisons be in a function that we call (like isArithmeticOper() for one of the cases you highlighted).

**Comment by AndyAyersMS** [comment]
In inline.h line 282 (266 before), this constructor's member init-list is messed up. Can we format-off here? Likewise in inline.cpp line 329  (326 before) and 707 (733 before). 

My preference would be to leave all the constructor member-init-lists as they were, but these three look especially odd -- likely since clang-format's preference for trailing commas interacts poorly with conditionally defined data members.

Rest of the changes in the inlining files look ok.

```
    InlinePolicy(bool isPrejitRoot)
        : m_Decision(InlineDecision::UNDECIDED), m_Observation(InlineObservation::CALLEE_UNUSED_INITIAL),
          m_IsPrejitRoot(isPrejitRoot)
#if defined(DEBUG) || defined(INLINE_DATA)
          ,
          m_IsDataCollectionTarget(false)
```

File: src/jit/alloc.cpp (hunk 1 above)
Comment (mikedn) [inline src/jit/alloc.cpp]: But can't we at least tell it not to reformat? VC++'s formatter has no trouble with these, it only fixes indentation if it's messed up.

File: src/jit/alloc.cpp (hunk 2 above)
Comment (mikedn) [inline src/jit/alloc.cpp]: So I take it that this is some kind of bug, these are not declarations.

File: src/jit/alloc.cpp (hunk 1 above)
Comment (adiaaida) [inline src/jit/alloc.cpp]: Yes, we can introduce // clang-format off // clang-format on blocks, but there are many instances of this change, and I was hoping to not pollute the codebase with clang comments.

File: src/jit/alloc.cpp (hunk 2 above)
Comment (adiaaida) [inline src/jit/alloc.cpp]: Sorry, I meant assignments.

There are two options: AlignConsecutiveAssignements and AlignConsecutiveDeclarations. We have both set to true.

**Comment by adiaaida** [comment]
@AndyAyersMS if I change BreakConstructorInitializersBeforeComma to true, it will produce this:

>      InlinePolicy(bool isPrejitRoot)
>         : m_Decision(InlineDecision::UNDECIDED)
>         , m_Observation(InlineObservation::CALLEE_UNUSED_INITIAL)
>         , m_IsPrejitRoot(isPrejitRoot)
> #if defined(DEBUG) || defined(INLINE_DATA)
>         , m_IsDataCollectionTarget(false)
> #endif // defined(DEBUG) || defined(INLINE_DATA)

Is that better?

File: src/jit/alloc.cpp (hunk 1 above)
Comment (adiaaida) [inline src/jit/alloc.cpp]: I can change ConstructorInitializerAllOnOneLineOrOnePerLine to true (right now it's false). I think in many cases, it will still do this all on the same time thing, but in some of them it may fix it.

File: src/inc/palclr.h (hunk 7 above)
Comment (RussKeldorph) [inline src/inc/palclr.h]: @janvorli The overarching motivation is to get clang to compile the code on Windows so clang-format/-tidy can run on Windows.  It's possible we may be able to tunnel the right compatibility flags through to Clang to get this to be accepted, but I believe we've tried that and failed.  If we can't, we need some workaround here.  In any case, these workarounds' conditions should only trigger in the problem formatting scenarios.  What is the "right" test to detect a Windows build?

Comment (janvorli) [inline src/inc/palclr.h]: The clang option `-fms-extensions` enables this `__declspec`. We use it for Unix builds. As for your question on the right test to detect a windows build, you can use PLATFORM_UNIX define.

Comment (adiaaida) [inline src/inc/palclr.h]: I get the following error without this change:

C:/michelm/coreclr/src/jit/_typeinfo.h:62:1: error: 'selectany' can only be applied to data items with external linkage [clang-diagnostic-error]
SELECTANY const ti_types g_jit_types_map[] = {
^
C:/michelm/coreclr/src/jit/../inc/palclr.h:487:37: note: expanded from macro 'SELECTANY'
#define SELECTANY extern __declspec(selectany). I'll try with PLATFORM_UNIX defined.

Comment (adiaaida) [inline src/inc/palclr.h]: PLATFORM_UNIX causes significantly more errors.

**Comment by CarolEidt** [comment]
I prefer what's produced with `BreakConstructorInitializersBeforeComma` set to `true`, as you show above. I think it would address several other cases which IMO have become less readable.

File: src/inc/palclr.h (hunk 7 above)
Comment (RussKeldorph) [inline src/inc/palclr.h]: @adiaaida The proposal is to add !PLATFORM_UNIX to the condition, not to define it during the translation.  At any rate, as we discussed offline, perhaps it makes sense to define a specific macro *only* during formatting to make the intent clear.

Comment (adiaaida) [inline src/inc/palclr.h]: @janvorli I could introduce a define SOURCE_FORMATTING and change the ifdef to use that. Would that be better?

**Comment by CarolEidt** [comment]
I've looked at all the (RyuJIT) backend sources, the SIMD sources, and a few others. While it's not ideal, it's certainly an improvement - and should streamline our reviews in future. I would like to see the `BreakConstructorInitializersBeforeComma` change, but otherwise LGTM.
It would be good to update the coding conventions document, and to mention there that we are using formatting tools to enforce many of the purely formatting conventions.

**Comment by adiaaida** [comment]
@CarolEidt I plan on switching on BreakConstructorInitializersBeforeComma and ConstructorInitializerAllOnOneLineOrOnePerLine. I have the change, but I'm waiting on @janvorli's ok for the change to use SOURCE_FORMATTING as a #define before pushing the changes.

File: src/inc/palclr.h (hunk 7 above)
Comment (janvorli) [inline src/inc/palclr.h]: It is strange that we don't get this error while compiling on Linux from other places where we have the same definition of SELECTANY and use it for const arrays. I would like to eventually investigate it, but not for this change.
Anyways, the palclr.h is not used on Unix at all - its contents is all wrapped in  `#ifndef FEATURE_PAL`, which is defined for Unix only, since only Unix uses PAL. So using the __clang__ define here is actually ok.

Comment (adiaaida) [inline src/inc/palclr.h]: We've introduced one in pal/inc/pal.h, clrtypes.h, and winwrap.h as well. Are all of those ok too?

Comment (janvorli) [inline src/inc/palclr.h]: Ah, I've missed that. No, those will need to be defined as they used to be for Unix, so the SOURCE_FORMATTING define introduction sounds like a good thing to do.
